dds_web/api/project.py

```diff
@@ -522,20 +518,47 @@ def archive_project(
                     "Please abort the project if you wish to proceed."
                 )
 
+        # Get redis connection to add a job if needed
+        redis_url = flask.current_app.config.get("REDIS_URL")
+        r = Redis.from_url(redis_url)
+        q = Queue(connection=r)
+
+        job_delete_contents = q.enqueue(
+            self.queue_helper_fun_delete_contents,
+            project_id=project.public_id,  # It is not possible to pass the project object directly to the queue
+            clear_proj_info=aborted,  # If aborted, clear project info
+        )
+
+        job_update_db = q.enqueue(
+            self.queue_helper_fun_update_db,
+            project_id=project.public_id,
+            current_time=current_time,
+            new_status="Archived",
+            aborted=aborted,
+            depends_on=job_delete_contents,  # This job is only executed after success of delete contents
+        )
+
+        return None, ""  # Dummy returns to not break the main function
+
+    @dbsession
+    def queue_helper_fun_delete_contents(self, project_id: int, clear_proj_info: bool = False):
+        """Delete the project contents for the archiving operation.
+        Function to be called by the queue."""
+
+        project = models.Project.query.filter_by(public_id=project_id).one_or_none()
+
         try:
             # Deletes files (also commits session in the function - possibly refactor later)
             RemoveContents().delete_project_contents(
                 project_id=project.public_id, delete_bucket=True
             )
-            delete_message = f"\nAll files in {project.public_id} deleted"
             self.rm_project_user_keys(project=project)
 
             # Only mark as inactive after all deletion operations succeed
             project.is_active = False
             # Delete metadata from project row
-            if aborted:
+            if clear_proj_info:
                 project = self.delete_project_info(project)
```

Comment:
This is up for discussion but I feel like this should be included in the database update function instead? If the  `queue_helper_fun_update_db` should only be for the status change then we should change that name to something else